sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -60,18 +60,24 @@ case class CreateArray(children: Seq[Expression]) extends Expression {
         ctx.INPUT_ROW,
         children.zipWithIndex.map { case (e, i) =>
           val eval = e.genCode(ctx)
-          eval.code + s"""
-            if (${eval.isNull}) {
```

Comment:
Let me change my opinion. Another PR will handle this elimination and related features.
This PR will focus on "each element of project_values is null if it never occurs".